lib/compilers/csharp.js

```diff
@@ -0,0 +1,151 @@
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+import path from 'path';
+
+import fs from 'fs-extra';
+
+import { BaseCompiler } from '../base-compiler';
+
+export class CSharpCompiler extends BaseCompiler {
+
+    static get key() { return 'csharp'; }
+
+    get rID() { return 'linux-x64'; }
+    get targetFramework() { return 'net6.0'; }
+    get buildConfig() { return 'Release'; }
+
+    get compilerOptions() {
+        return ['publish', '-c', this.buildConfig, '--self-contained', '--runtime', this.rID];
+    }
+
+    async runCompiler(compiler, options, inputFileName, execOptions) {
+        if (!execOptions) {
+            execOptions = this.getDefaultExecOptions();
+        }
+
+        const programDir = path.dirname(inputFileName);
+        const clrBuildDir = '/home/ubuntu/runtime/artifacts/bin/coreclr/Linux.x64.Checked';
+
+        const projectFilePath = path.join(programDir, 'Program.csproj');
+        const coreClrJitPath = path.join(clrBuildDir, 'libclrjit.so');
+        const crossgenPath = path.join(clrBuildDir, 'crossgen');
+
+        const programPublishPath = path.join(
+            programDir,
+            'bin',
+            this.buildConfig,
+            this.targetFramework,
+            this.rID,
+            'publish',
+        );
+
+        const programDllPath = path.join(programPublishPath, 'Program.dll');
+
+        const projectFileContent =
+        `<Project Sdk="Microsoft.NET.Sdk">
+            <PropertyGroup>
+         <TargetFramework>${this.targetFramework}</TargetFramework>
+         <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+         </PropertyGroup>
+         </Project>
+        `;
+
+        execOptions.customCwd = programDir;
+        await fs.writeFile(projectFilePath, projectFileContent);
+        options = this.compilerOptions;
+
+        const compilerResult = await super.runCompiler(
+            compiler,
+            options,
+            inputFileName,
+            execOptions,
+        );
+
+        if (compilerResult.code !== 0)
+            return compilerResult;
+
+        const crossgenResult = await this.runCrossgen(
+            execOptions,
+            crossgenPath,
+            coreClrJitPath,
+            programPublishPath,
+            programDllPath,
+            this.getOutputFilename(programDir, ''),
+        );
+
+        if (crossgenResult.code !== 0)
+            return crossgenResult;
+
+        return compilerResult;
+    }
+
+    optionsForFilter() {
+        return this.compilerOptions;
+    }
+
+    getOutputFilename(dirPath) {
+        return path.join(dirPath, `output.s`);
+    }
+
+    cleanAsm(asm) {
+        var cleanedAsm = '';
+        var asmLines = asm.split('\n');
+
+        for (const line of asmLines) {
+            if (line.startsWith('; Assembly listing for method')) {
+                cleanedAsm = cleanedAsm.concat(line.substring(30) + ':\n');
```

Comment:
another minor nit, the magic number 30 is significant because?

giving it a name will add context for noobs like myself.